pkg/registry/resource/devicetaintrule/strategy.go

```diff
@@ -34,51 +36,104 @@ type deviceTaintRuleStrategy struct {
+func (*deviceTaintRuleStrategy) WarningsOnCreate(ctx context.Context, obj runtime.Object) []string {
 	return nil
 }
 
-func (deviceTaintRuleStrategy) Canonicalize(obj runtime.Object) {
+func (*deviceTaintRuleStrategy) Canonicalize(obj runtime.Object) {
 }
 
-func (deviceTaintRuleStrategy) AllowCreateOnUpdate() bool {
+func (*deviceTaintRuleStrategy) AllowCreateOnUpdate() bool {
 	return false
 }
 
-func (deviceTaintRuleStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
-	patch := obj.(*resource.DeviceTaintRule)
-	oldPatch := old.(*resource.DeviceTaintRule)
+func (*deviceTaintRuleStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	rule := obj.(*resource.DeviceTaintRule)
+	oldRule := old.(*resource.DeviceTaintRule)
+	rule.Status = oldRule.Status
 
 	// Any changes to the spec increment the generation number.
-	if !apiequality.Semantic.DeepEqual(oldPatch.Spec, patch.Spec) {
-		patch.Generation = oldPatch.Generation + 1
+	if !apiequality.Semantic.DeepEqual(oldRule.Spec, rule.Spec) {
+		rule.Generation = oldRule.Generation + 1
 	}
 }
 
-func (deviceTaintRuleStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
+func (*deviceTaintRuleStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
 	return validation.ValidateDeviceTaintRuleUpdate(obj.(*resource.DeviceTaintRule), old.(*resource.DeviceTaintRule))
 }
 
-func (deviceTaintRuleStrategy) WarningsOnUpdate(ctx context.Context, obj, old runtime.Object) []string {
+func (*deviceTaintRuleStrategy) WarningsOnUpdate(ctx context.Context, obj, old runtime.Object) []string {
 	return nil
 }
 
-func (deviceTaintRuleStrategy) AllowUnconditionalUpdate() bool {
+func (*deviceTaintRuleStrategy) AllowUnconditionalUpdate() bool {
 	return true
 }
+
+type deviceTaintRuleStatusStrategy struct {
+	*deviceTaintRuleStrategy
+}
+
+// GetResetFields returns the set of fields that get reset by the strategy and
+// should not be modified by the user. For a status update that is the spec.
+func (*deviceTaintRuleStatusStrategy) GetResetFields() map[fieldpath.APIVersion]*fieldpath.Set {
+	fields := map[fieldpath.APIVersion]*fieldpath.Set{
+		"resource.k8s.io/v1alpha3": fieldpath.NewSet(
+			fieldpath.MakePathOrDie("spec"),
+		),
+	}
+
+	return fields
+}
+
+func (*deviceTaintRuleStatusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newRule := obj.(*resource.DeviceTaintRule)
+	oldRule := old.(*resource.DeviceTaintRule)
+	newRule.Spec = oldRule.Spec
+	metav1.ResetObjectMetaForStatus(&newRule.ObjectMeta, &oldRule.ObjectMeta)
+}
+
+func (r *deviceTaintRuleStatusStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
+	newRule := obj.(*resource.DeviceTaintRule)
+	oldRule := old.(*resource.DeviceTaintRule)
+	return validation.ValidateDeviceTaintRuleUpdate(newRule, oldRule)
```

Comment:
since this can only change fields under status, possibly make a distinct `ValidateDeviceTaintRuleStatusUpdate` method that only checks stuff under status? anything invalid under spec cannot be fixed via a status call anyway